src/mongo_types.ts

```diff
@@ -99,8 +100,15 @@ export interface RootFilterOperators<TSchema> extends Document {
   $comment?: string | Document;
 }
 
+/**
+ * A type that extends Document but forbids anything that "looks like" an object id.
+ */
+export type NonObjectIdLikeDocument = {
+  [key in keyof ObjectIdLike]?: never
+} & Document
+
 /** @public */
-export interface FilterOperators<TValue> extends Document {
+export interface FilterOperators<TValue> extends NonObjectIdLikeDocument {
```

Comment:
The reason we can move forward with this is because the chance of running into a type error is very low here right?

So if I have a schema-parameterized collection. And I attempt to use the keys: 
- `id`
- `__id`
- `toHexString`

And for whatever reason these keys DON'T appear in my schema, then I will have issues. But given that these three specific keys are not `_id` it is very unlikely that they would both be in use and omitted from a schema definition.

Does that summary sound accurate?